### PR TITLE
CircleCI take over Travis

### PR DESCRIPTION
**Pull request by dincho**
Prove can be found on: [CircleCI successful build](https://circleci.com/workflow-run/e2ebad21-7857-4b8d-9fa6-96dc17090d9e) and [GitHub release assets](https://github.com/dincho/epoch/releases) of my fork.

[PT 154999652](https://www.pivotaltracker.com/story/show/154999652)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,6 +82,16 @@ references:
             --extra-vars "package=${PACKAGE:?} env=${DEPLOY_ENV:?}" \
             deploy.yml
 
+  upload_github_release_asset: &upload_github_release_asset
+    run:
+      name: Upload GitHub Release Asset
+      environment:
+        PACKAGES_DIR: *packages_workspace
+      command: |
+        ./deployment/upload-github-release-assets.sh github_api_token=${GITHUB_API_TOKEN:?} \
```
Comment (lucafavatella) [inline .circleci/config.yml]: I assume no spaces possible hence no quoting.

File: .circleci/config.yml [hunk 2]
```diff
@@ -78,6 +82,16 @@ references:
...
+          owner=${CIRCLE_PROJECT_USERNAME} repo=${CIRCLE_PROJECT_REPONAME} \
+          tag=${CIRCLE_TAG} ASSETS=${PACKAGES_DIR:?}/*
```
Comment (lucafavatella) [inline .circleci/config.yml]: Assumes no spaces in tag. Reasonable.

File: .travis.yml [hunk 3]
```diff
@@ -1,63 +0,0 @@
...
-  - ./ci install "${JOB:?}" "${TRAVIS_BUILD_DIR:?}"
-script:
-  - ./ci script "${JOB:?}" "${TRAVIS_BUILD_DIR:?}"
-after_failure:
-  - ./ci after_failure "${JOB:?}" "${TRAVIS_BUILD_DIR:?}"
-deploy:
-  - provider: releases
-    api_key:
-      # See https://docs.travis-ci.com/user/deployment/releases/#Authenticating-with-an-OAuth-token
-      secure: SsNvB5u94rozdGDwn4jhkbR9RoN/KQQYij1vXiJzmk1y997GsLG/skBlgqmpuGQap+RgvDHWAJ0sxZh0H+xHZB78SKmhryvLdPTmuN4B3qgvIuJPDtvrogtEDvZ8SXQXOXPlH/o7hBvyfebUXDIqn8UU8DUaexnpmwgxjI+AezHIFRtwyd29pDBQHLebHo3OoB6hsKM6mqxb/SfDXhktSfoqlZ2zQvU1boXzwMo/S54OMY6z2Ui0xbdXilkxwki8yW88ouHsyekmErYMGXS/j1gubxDkMQ+ZO+BcbFR4ZRt2a8FUkqr2pymzf4lBACkxSzzSKJrC8Y49ENpqsCHiZE1ATNsoNKMn8XNynf/N4H6lc8np3BOdcBA+lGz40CAZI8kMal+pGXKsex7lbbvs/gkNufD9LCiW4JAkA/ndFPYobuxE07nraSvD/b8sP2H5uSMXJWMwZMgudmUZh2kdwZFCFTyIAbk8AC94wpJFII/45rL9yXf4tfbBqTUgPnGs0cUuKCRPkdQ5uNnHNzSpi/CQhHob3JVXBJ1Ve67SiUDU44kvkv6PdnX1i95wJkcPeX485amW0w2qX3CwGrwm9pnoOZTRkfBXMRitfxQ2i42r6T7zZTgKjPHDg8Stf6ZPqbmTLj4++hni8bGNICs5caZKinKCH3e60PL5JzPZ0u8=
```
Comment (lucafavatella) [inline .travis.yml]: After this PR is merged we shall delete any traces of this token before testing deployment to UAT.

File: ci [hunk 4]
```diff
@@ -1,88 +0,0 @@
...
-        # speed up mining during the tests.
-        make python-env
-        make python-release-test WORKDIR="$(mktemp -d)" TARBALL="${FinalPackagePath:?}"
-        ;;
-    after_failure-*)
-        LogsDir="${BuildDir}/_build/test/logs"
-        echo "logs/ct.latest.log\n"
-        cat ${LogsDir}/ct.latest.log
-        ls -l ${LogsDir}/latest.sync
-        for f in `ls -1 ${LogsDir}/ct_run*/lib.*.logs/run*/suite.log`; do escript failed_logs.escript $f; done
```
Comment (lucafavatella) [inline ci]: failed_logs.escript could be deleted contextually to this PR.

File: .circleci/config.yml (hunk 1 above)
Comment (dincho) [reply .circleci/config.yml]: yes

File: ci (hunk 4 above)
Comment (dincho) [reply ci]: deleted in 84cdea4

**Pull request merged**